sympy/functions/elementary/exponential.py

```diff
@@ -436,11 +436,27 @@ def _eval_nseries(self, x, n, logx):
         if arg0 in [-oo, oo]:
             return self
         t = Dummy("t")
-        exp_series = exp(t)._taylor(t, n)
+        nterms = n
+        try:
+            cf = Order(arg.as_leading_term(x), x).getn()
+        except NotImplementedError:
+            cf = 0
+        if cf > 0:
```

Comment:
try should also cover `if cf > 0` otherwise it will fail beacuse `None can be comapred with 0(if cf != 0) but can not be judged whether it is greater than 0 or not.

Reply:
Maybe `if cf and cf > 0'. That should handle the None case.

Reply:
Yeah...You are right it should work.